ql/src/java/org/apache/hadoop/hive/ql/parse/BaseSemanticAnalyzer.java

```diff
@@ -1707,7 +1707,12 @@ public static Map<String, String> getPartSpec(ASTNode node) {
   public static void validatePartSpec(Table tbl, Map<String, String> partSpec,
       ASTNode astNode, HiveConf conf, boolean shouldBeFull) throws SemanticException {
     if (tbl.getStorageHandler() != null && tbl.getStorageHandler().alwaysUnpartitioned()) {
-      tbl.getStorageHandler().validatePartSpec(tbl, partSpec);
+      if (Context.RewritePolicy.fromString(conf.get(HiveConf.ConfVars.REWRITE_POLICY.varname)) == 
+          Context.RewritePolicy.PARTITION) {
+        tbl.getStorageHandler().validatePartAnySpec(tbl, partSpec);
```

Comment:
wouldn't it be better to move if logic inside of `StorageHandler.validatePartSpec()` ?

Reply:
Yes; Done